### ISSUE TITLE
ClassNotFoundException: org.objectweb.asm.commons.Remapper

### ISSUE THREAD

**Issue by bloodmc** [johnrengelman/shadow#29]
I built the plugin using the latest source with 'build gradle'. I then proceeded to run shadow on my project as shown in the log below.
Here is the full log http://pastie.org/8598403

If I manually add the objectweb lib to the shadow jar, it works fine. Not sure if I built it incorrectly, let me know. Thanks

**Comment by cjstehno** [comment]
Ugh... just ran into this exact issue (Gradle 2.13, Shadow: 1.2.3). I was able to work around the problem by specifying the explicit dependency in the `buildscript` block `classpath 'org.ow2.asm:asm-debug-all:5.0.4'` - I ran into another classpath collision but got passed this one. It seems that the shadow plugin and the jacoco plugin do not play nice together at all. If I come across a better solution, I will post it here.

**Comment by johnrengelman** [comment]
@cjstehno - can you post your buildscript block?

**Comment by cjstehno** [comment]
Here is what finally got through all the `ClassNotFoundException`s:

```
buildscript {
    repositories {
        mavenLocal()
        maven { url prodReleaseRepo }
        maven { url prodSnapshotRepo }
    }
    dependencies {
        // jacoco and shadow are at odds about dependencies
        classpath 'org.apache.ant:ant:1.9.3'
        classpath 'org.ow2.asm:asm-debug-all:5.0.4'
        classpath 'commons-io:commons-io:2.4'

        classpath 'com.github.jengelman.gradle.plugins:shadow:1.2.3'
    }
}
```

Notice that we are not allowed to use `jcenter()` which was the original trigger of the problem - it seems that adding that back in would make it work, but it wasn't an option when stuck inside the company artifactory instance.

Also note, that we are using the standard jacoco plugin from Gradle 2.13.

**Comment by bohannan** [comment]
+1 I was getting:

> Failed to apply plugin [class 'com.github.jengelman.gradle.plugins.shadow.ShadowJavaPlugin']
   > Could not create task of type 'ShadowJar'.

When I looked into it a little bit more I saw that I was getting several ClassNotFoundExceptions, same solution as above resolved the issue. However, we are not using jacoco plugin in our project (Gradle 2.9).

**Comment by johnrengelman** [comment]
This should be addressed in the 2.0.0 release by shadowing all of Shadow's dependencies to a new package location.
